app/modules/ads1115.c

```diff
@@ -581,11 +557,23 @@ static const LUA_REG_TYPE ads1115_map[] = {
     {   LSTRKEY( "COMP_1CONV" ),    LNUMVAL(ADS1115_CQUE_1CONV)     },
     {   LSTRKEY( "COMP_2CONV" ),    LNUMVAL(ADS1115_CQUE_2CONV)     },
     {   LSTRKEY( "COMP_4CONV" ),    LNUMVAL(ADS1115_CQUE_4CONV)     },
-    {   LSTRKEY( "ADS1015"),        LNUMVAL(ADS1115_ADS1015)        },
-    {   LSTRKEY( "ADS1115"),        LNUMVAL(ADS1115_ADS1115)        },
     {   LSTRKEY( "CMODE_TRAD"),     LNUMVAL(ADS1115_CMODE_TRAD)     },
     {   LSTRKEY( "CMODE_WINDOW"),   LNUMVAL(ADS1115_CMODE_WINDOW)   },
     {   LNILKEY, LNILVAL                                            }
```

Comment:
The `__gc` index is not defined - I suspect memory leaks when the device object is disposed by the garbage collector while `timer_ref` still holds a reference to the callback. Similarly, an armed os_timer would need to be shut down gracefully.

Reply:
Somehow I convinced myself that it will not happen, but on the second iteration I believe it will be worse than memory leak.
1. Timer is armed.
2. Device object is garbage collected.
3. Timer fires, with ads_ctrl referencing user data which can be anything at this moment.